packages/api/src/utils/psa.js

```diff
@@ -204,8 +205,29 @@ export function validateSearchParams (queryString) {
     opts.limit = DEFAULT_PIN_LISTING_LIMIT
   }
 
+  if (meta) {
+    // Must be a string representation of a JSON object.
+    let metaJson
+    try {
+      metaJson = JSON.parse(meta)
+    } catch (e) {
+      return {
+        error: new PSAErrorInvalidData(INVALID_META),
+        data: undefined
+      }
+    }
+
+    if (Object.entries(metaJson).some(([, v]) => typeof v !== 'string')) {
```

Comment:
Do we need this validation? Based [on nft's code](https://github.com/nftstorage/nft.storage/blob/main/packages/api/src/routes/pins-list.js#L69) it doesn't seem like meta values need to be strings (I know it used to be like that originally, but I don't think we've ever confirmed it)

If the values/keys do need to be strings, I think it can be achieved by adding something like this to the validator:
```
    meta: {
      type: 'object',
      keys: { type: 'string' },
      values: { type: 'string' }
    },
```
But I'd test this to make sure ^

Reply:
Good suggestion, have added to the validator to enforce string type values. 

`JSON.parse()` will validate for string value keys.